src/Majora/Framework/Loader/LoaderTrait.php

```diff
@@ -49,14 +56,60 @@ public function setUp(
         $collectionClass,
         RepositoryInterface $entityRepository = null
     ) {
-        $this->entityClass      = $entityClass;
+        @trigger_error(__METHOD__.'() is deprecated and will be removed in 2.0. Please use configureMetadata() instead.', E_USER_DEPRECATED);
+
+        if ($entityRepository) {
+            @trigger_error('Repository injection throught '.__METHOD__.'() is deprecated and will be removed in 2.0. Please inject it by constructor.', E_USER_DEPRECATED);
+
+            $this->entityRepository = $entityRepository;
+        }
+
+        return $this->configureMetadata($entityClass, $entityProperties, $collectionClass);
+    }
+
+    /**
+     * Checks if loader is properly configured.
+     *
+     * @throws \RuntimeException if not configured
+     */
+    private function assertIsConfigured()
+    {
+        if ($this->entityRepository && $this->entityClass
+            && $this->collectionClass && $this->filterResolver
+        ) {
```

Comment:
you're requiring `entityRepository` here but it's defined as optional in both `LoaderTrait::setUp()` and `DoctrineLoaderTrait::__construct()`